apps/worker/src/app/workflow/services/metric-queue.service.ts

```diff
@@ -0,0 +1,125 @@
+const nr = require('newrelic');
+import { WorkerOptions } from 'bullmq';
+import { Inject, Injectable, Logger } from '@nestjs/common';
+import { QueueService } from '@novu/application-generic';
+import { min, max, sum, mean } from 'simple-statistics';
+
+interface IMetric {
+  count: number;
+  total: number;
+  min: number;
+  max: number;
+  sumOfSquares: number;
+}
+
+function sumOfSquares(values: number[]): number {
+  const meanVal = mean(values);
+
+  return sum(values.map((value) => (value - meanVal) ** 2));
+}
+
+@Injectable()
+export class MetricQueueService extends QueueService<Record<string, never>> {
+  constructor(@Inject('BULLMQ_LIST') private token_list: QueueService[]) {
+    super('metric');
+
+    Logger.warn('Metric queue service created');
+
+    this.bullMqService.createWorker(this.name, this.getWorkerProcessor(), this.getWorkerOpts());
+
+    this.bullMqService.worker.on('completed', async (job) => {
+      await this.jobHasCompleted(job);
+    });
+
+    this.bullMqService.worker.on('failed', async (job, error) => {
+      await this.jobHasFailed(job, error);
+    });
+
+    void this.addToQueue('metric-job', {}, '', {
+      repeat: {
+        immediately: true,
+        pattern: '* * * * * *',
+      },
+    });
+  }
+
+  public async gracefulShutdown() {
+    // Right now we only want this for testing purposes
+    if (process.env.NODE_ENV === 'test') {
+      await this.bullMqService.queue.drain();
+      await this.bullMqService.worker.close();
+    }
+  }
+
+  private getWorkerOpts(): WorkerOptions {
+    return {
+      ...this.bullConfig,
+      lockDuration: 90000,
+      concurrency: 200,
+      settings: {},
+    } as WorkerOptions;
+  }
+
+  private getWorkerProcessor() {
+    return async () => {
+      return await new Promise(async (resolve, reject) => {
+        for (const queueService of this.token_list) {
+          this.bullMqService.createQueue(queueService.name, {
```

Comment:
This creates a global instance on the parent class, and can lead to a lot of weird behaviour. We should create a new bullMq service instance for each Queue we have here.